extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnRecorder.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.security.webauthn;
+
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.vertx.http.runtime.HttpConfiguration;
+import io.quarkus.vertx.http.runtime.security.PersistentLoginManager;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BodyHandler;
+
+@Recorder
+public class WebAuthnRecorder {
+
+    private static final Logger log = Logger.getLogger(WebAuthnRecorder.class);
+
+    final RuntimeValue<HttpConfiguration> httpConfiguration;
+    final RuntimeValue<WebAuthnRunTimeConfig> config;
+
+    //the temp encryption key, persistent across dev mode restarts
+    static volatile String encryptionKey;
+
+    public WebAuthnRecorder(RuntimeValue<HttpConfiguration> httpConfiguration, RuntimeValue<WebAuthnRunTimeConfig> config) {
+        this.httpConfiguration = httpConfiguration;
+        this.config = config;
+    }
+
+    public void setupRoutes(BeanContainer beanContainer, RuntimeValue<Router> routerValue) {
+        WebAuthnSecurity security = beanContainer.instance(WebAuthnSecurity.class);
+        WebAuthnAuthenticationMechanism authMech = beanContainer.instance(WebAuthnAuthenticationMechanism.class);
+        IdentityProviderManager identityProviderManager = beanContainer.instance(IdentityProviderManager.class);
+        WebAuthnController controller = new WebAuthnController(security, config.getValue(), identityProviderManager, authMech);
+        Router router = routerValue.getValue();
+        BodyHandler bodyHandler = BodyHandler.create();
+        // FIXME: paths configurable
```

Comment:
I'm guessing these paths should start with `/q` perhaps?

Reply:
Steph, can you explain a bit more about these routes - are they all protected, or can a user without a WebAuthn registration invoke on any of these routes ?

Reply:
If they are framework routes, they should probably under `/q`

Reply:
Yeah, I thought the same.

Reply:
This path, `q` or something else, is configurable, I don't recall the name, something with the `non-application` qualifier

Reply:
See `NonApplicationRootPathBuildItem`

Reply:
@FroMage 

> These routes are not protected: they're the only way in to authenticate and register via webauthn.
But did you explain earlier `WebAuthnAuthenticationMechanism` was an alternative to some of these autonomous routes ? 

and re the `/register` route - (I read the docs but likely missed something) where is that public key saved ? The docs says it is not saved anywhere, so I just wonder how is the signature verified.

Reply:
Done for the `/q` prefix

Reply:
@FroMage I suppose I was confused by `So the association of the hardware authenticator and the target service will always derive the same private and public key pair, none of which are stored anywhere`, but indeed the docs describe it, sorry

Reply:
@FroMage 

> WebAuthnAuthenticationMechanism is just an implementation detail.

I don't think it is a correct statement - it implements `HttpAuthenticationMechanism` - which means it should drive the authentication on its own without the user having to start from `/webauthn/login`. I.e, if we have 
```
@Path("myservice")
class MyEndpoint { @GET String get() {return "hello";} }
```
then `WebAuthnAuthenticationMechanism` should drive the authentication when a user accesses `http://localhost:8080/myservice`. I'm looking at the code and it appears to be the case - i.e, it challenges the user with a redirect to `/login`

Reply:
@FroMage, Steph, I must be slow, sorry, so given this flow I've just described, a few questions:
- do you expect it to work the way I've described
- what happens to the original `http://localhost:8080/myservice` request - should the user be expected to be returned to `http://localhost:8080/myservice` ? (I don't see it is a redirect back in the `/login` handler, not a bit problem, just curious)

Reply:
Yes, it should redirect you to the login page, which the user has to implement.

And no, it doesn't redirect you back to the original page ATM, thought I guess we could add support for that in the future, if someone asks for it.

Reply:
@FroMage OK, thanks. I think `WebAuthn` is a better `Form Auth` - it basically has the same kind of flow, with the main thing which is different is no password is sent.